src/core/index.js

```diff
@@ -369,18 +369,25 @@ util.extend( corefn, {
           }
         }
 
+        let filterElesToRemove = function (eles) {
+          return eles.stdFilter( function( ele ){
+              return !idInJson[ ele.id() ];
+          }).toArray();
```

Comment:
Better to use the collection than an array

Reply:
i used array and merge, because I can't be sure of the order in a collection, as I'm in a for-loop I rely on the order so i ensure that the moved objects are in the end. I'm using it as a "queue", but not removing any element of it, just incrementing the index.